library/src/doo/runner.cljs

```diff
@@ -51,12 +51,20 @@
 
 ;; Karma starts the runner with arguments
 (defn ^:export run! [a]
-  (try
-    (*main-cli-fn* a)
-    (catch :default e
-      (println "WARNING: doo's init function was not set")
-      (println e)
-      (exit! false))))
+  (if-not (fn? *main-cli-fn*)
+    (do (println "WARNING: doo's init function was not set")
+        (exit! false))
+    (try
+      (*main-cli-fn* a)
+      (catch :default e
+        (println "\nERROR: Exception outside tests:")
+        (println e)
+        (println "\nERROR: Stacktrace:")
+        (println
+         (if (.hasOwnProperty e "stack")
+           (.-stack e)
+           "No stacktrace available."))
```

Comment:
I think that in case there's no stacktrace, you should either see `No stacktrace available` or nothing at all, but not `ERROR: Stacktrace:`. I.e. please move the printing inside the if block, as you said.